vmdb/app/models/ems_refresh/parsers/openstack_infra.rb

```diff
@@ -139,22 +139,24 @@ def parse_stack_parameter(param_key, param_val, stack_id)
       def parse_host(host, indexed_servers, _indexed_hosts_ports, indexed_resources)
         uid = host.uuid
         host_name = identify_host_name(indexed_resources, host.instance_uuid, uid)
+        ip_address = identify_primary_ip_address(host, indexed_servers)
 
         new_result = {
-          :name             => host_name,
-          :type             => 'HostOpenstackInfra',
-          :uid_ems          => uid,
-          :ems_ref          => uid,
-          :ems_ref_obj      => host.instance_uuid,
-          :operating_system => {:product_name => 'linux'},
-          :vmm_vendor       => 'RedHat',
-          :vmm_product      => identify_product(indexed_resources, host.instance_uuid),
-          :ipaddress        => identify_primary_ip_address(host, indexed_servers),
-          :mac_address      => identify_primary_mac_address(host, indexed_servers),
-          :ipmi_address     => identify_ipmi_address(host),
-          :power_state      => lookup_power_state(host.power_state),
-          :connection_state => lookup_connection_state(host.power_state),
-          :hardware         => process_host_hardware(host)
+          :name                => host_name,
+          :type                => 'HostOpenstackInfra',
+          :uid_ems             => uid,
+          :ems_ref             => uid,
+          :ems_ref_obj         => host.instance_uuid,
+          :operating_system    => {:product_name => 'linux'},
+          :vmm_vendor          => 'RedHat',
+          :vmm_product         => identify_product(indexed_resources, host.instance_uuid),
+          :ipaddress           => ip_address,
+          :hostname            => ip_address,
+          :mac_address         => identify_primary_mac_address(host, indexed_servers),
+          :ipmi_address        => identify_ipmi_address(host),
+          :power_state         => lookup_power_state(host.power_state),
+          :connection_state    => lookup_connection_state(host.power_state),
+          :hardware            => process_host_hardware(host),
```

Comment:
Kill the extra space between the keys and hash rockets here.  The minimum distance only needs to be one space from the longest key.